terraform/main.tf

```diff
@@ -0,0 +1,415 @@
+    }
+  }
+
+  tags = {
+    Terraform   = "true"
+    Createdby   = "Terraform"
+    Environment = "${local.environment}"
+  }
+}
+
+################################################################################
+# SG
+################################################################################
+resource "aws_security_group_rule" "allow_node_sg_to_cluster_sg" {
+  description = "Self-Node Group to Cluster API/MNG all traffic"
+
+  source_security_group_id = module.eks_blueprints.worker_node_security_group_id
+  security_group_id        = module.eks_blueprints.cluster_primary_security_group_id
+  type                     = "ingress"
+  protocol                 = "-1"
+  from_port                = 0
+  to_port                  = 0
+
+  depends_on = [
+    module.eks_blueprints
+  ]
+}
+
+resource "aws_security_group_rule" "allow_node_sg_from_cluster_sg" {
+  description              = "Cluster API/MNG to Self-Nodegroup all traffic"
+  source_security_group_id = module.eks_blueprints.cluster_primary_security_group_id
+  security_group_id        = module.eks_blueprints.worker_node_security_group_id
+  type                     = "ingress"
+  protocol                 = "-1"
+  from_port                = 0
+  to_port                  = 0
+
+  depends_on = [
+    module.eks_blueprints
+  ]
+}
+################################################################################
+# EKS Addons
+################################################################################
+module "eks_blueprints_kubernetes_addons" {
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.29.0"
+
+  eks_cluster_id       = module.eks_blueprints.eks_cluster_id
+  eks_cluster_endpoint = module.eks_blueprints.eks_cluster_endpoint
+  eks_oidc_provider    = module.eks_blueprints.oidc_provider
+  eks_cluster_version  = module.eks_blueprints.eks_cluster_version
+
+  # EKS Addons
+  enable_amazon_eks_vpc_cni            = true
+  enable_amazon_eks_coredns            = true
+  enable_amazon_eks_kube_proxy         = true
+  enable_amazon_eks_aws_ebs_csi_driver = true
+  enable_aws_efs_csi_driver            = true
+
+  #K8s Add-ons
+  enable_metrics_server               = true
+  enable_cluster_autoscaler           = true
+  enable_aws_load_balancer_controller = true
+
+  enable_argocd         = true
+  argocd_manage_add_ons = true
+  argocd_applications = {
+    addons = {
+      path               = "chart"
+      repo_url           = "https://github.com/aws-samples/eks-blueprints-add-ons.git"
+      add_on_application = true # Indicates the root add-on application.
+    }
+
+    // https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/argocd/main.tf#L94-L114
+    kustomize-apps = {
+      path            = "argocd-root"
+      repo_url        = "https://github.com/celestiaorg/testground-infra.git"
+      target_revision = "jose/hackground-k8s-tf" // Remove this one, should use main
```

Comment:
we need to prepare this to use testground/testground after this PR get's merged in out celestiaorg/testground and we do a proper PR to upstream

Reply:
exactly, I had the same idea, after we merge this PR, we'll be able to test it using the celestia repo, if everything works (it should), then, I'll prepare a new PR pointing to the final repo: [testground/infra](https://github.com/testground/infra)